File: avancado-intermediario/Java/Q3/Circulo.java
public class Circulo extends Figura {
    private double raio;

    public Circulo(double x, double y, double raio) {
        super(x, y);
        this.raio = raio;
        System.out.println("Criando círculo");
    }

    public void desenhar() {
        System.out.println("Desenhando um círculo na posição (" + getX() + ", " + getY() + ") com raio de " + raio);
    }
}

File: avancado-intermediario/Python/Q3 - Figura/circulo.py
from figura import Figura
class Circulo(Figura):

    def __init__(self,x:float,y:float,raio:float):
        super().__init__(x,y)
        print("Criando círculo...")
        self.__raio = raio
        
    @property
    def raio(self):
        return self.__raio

    @raio.setter
    def raio(self, novo_raio:float):
        self.__raio = novo_raio
        
    def desenhar(self):
        print(f"Desenhando um círculo na posição [{self.x},{self.y}] com raio de {self.__raio}")